kafka/consume.go

```diff
@@ -90,14 +95,45 @@ func (r *ConfluentConsumer) Seek(offset int64, timeout time.Duration) error {
 		return errors.New("Timeout should be set when calling Seek")
 	}
 
-	err := r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	var err error
+	if r.partition < 0 {
+		err = r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	} else {
+		err = r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Partition: r.partition, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	}
```

Comment:
Like this without the white space errors I have

```suggestion
       topicPartition := kafkalib.TopicPartition{
               Topic: &r.conf.Topic, 
               Offset: kafkalib.Offset(offset),
               }
	 if r.partition >= 0 {
             topicPartition.Partition = r.partition
	  } 
	  err := r.c.Seek(topicPartition, int(timeoutMs))
	}
```